next.config.js

```diff
@@ -2,6 +2,19 @@ const {
   PHASE_PRODUCTION_BUILD,
   PHASE_PRODUCTION_SERVER
 } = require('next/constants')
+const path = require('path')
+
+const webpack = (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
```

Comment:
What's this doing @bcomnes

Reply:
This just customizes the webpack config. 

https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config

In this case, its aliasing some peer dependencies to always resolve to top level (how peer deps should work).  

It works during linking, and also when you are not linking.

Reply:
Makes sense, i'm surprised this isn't the default behavior! Do you know what is?

Reply:
One of the crappy characteristics of peer deps and how it interacts with the node dependency resolution algorithm.  Packages that define peer deps are expected to be installed with the peer deps de-duplicated top level, but this doesn't happen with you link in a package with a full install that includes the peer dep in devDeps (as you would expect in a react package).

Reply:
More info: https://medium.com/@penx/managing-dependencies-in-a-node-package-so-that-they-are-compatible-with-npm-link-61befa5aaca7